libsolidity/formal/BMC.cpp

```diff
@@ -443,8 +445,10 @@ bool BMC::visit(ForStatement const& _node)
 		auto indices = copyVariableIndices();
 		if (_node.condition())
 		{
+			m_loopCheckpoints.emplace();
 			_node.condition()->accept(*this);
 			forCondition = expr(*_node.condition());
+			m_loopCheckpoints.pop();
```

Comment:
You're adding/moving these in 3 different places, but you're only adding one new test case and not updating any existing ones. I suspect that we don't have coverage for all 3.

The example I posted under the changelog would probably cover one of them. What's the other one?

Reply:
I've added you example with `while` loop, which covers the change in `visit(WhileStatement)`.
The one test with for loop that I added originally actually exercises both changes in `visit(ForStatement)`, so no additional test is needed.

Reply:
To clarify, there are two places in `visit(ForStatement)` where the loop condition is visited:
Before each unrolling, and a final check when the loop no longer enters another iteration.
We needed to patch both places.

Reply:
Thanks for clarification. Looks good now.